pkg/webhook/resources/virtualmachine/validator.go

```diff
@@ -530,3 +536,40 @@ func (v *vmValidator) checkReservedMemoryAnnotation(vm *kubevirtv1.VirtualMachin
 func (v *vmValidator) checkStorageResourceQuota(vm *kubevirtv1.VirtualMachine, oldVM *kubevirtv1.VirtualMachine) error {
 	return v.rqCalculator.CheckStorageResourceQuota(vm, oldVM)
 }
+
+func (v *vmValidator) storageClassOfVolumeChanged(oldVM, newVM *kubevirtv1.VirtualMachine) error {
```

Comment:
This functions looks almost identical to `checkResizeVolumes()`. Consider reconciling them by:

1. Moving the check criteria from line 562 to 568 to that function
2. Rename that function from `checkResizeVolumes()` to `checkUpdateAnnotations()` or something

I think your test cases will still pass with the change.